nifi/nifi-nar-bundles/nifi-flume-bundle/nifi-flume-processors/src/main/java/org/apache/nifi/processors/flume/FlumeSourceProcessor.java

```diff
@@ -0,0 +1,191 @@
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @OnScheduled
+    public void onScheduled(final SchedulingContext context) {
+        try {
+            source = SOURCE_FACTORY.create(
+                    context.getProperty(SOURCE_NAME).getValue(),
+                    context.getProperty(SOURCE_TYPE).getValue());
+
+            String flumeConfig = context.getProperty(FLUME_CONFIG).getValue();
+            String agentName = context.getProperty(AGENT_NAME).getValue();
+            String sourceName = context.getProperty(SOURCE_NAME).getValue();
+            Configurables.configure(source,
+                    getFlumeSourceContext(flumeConfig, agentName, sourceName));
+
+            if (source instanceof EventDrivenSource) {
+                runner = new EventDrivenSourceRunner();
+                channel = new MemoryChannel();
+                Configurables.configure(channel, new Context());
+                channel.start();
+                source.setChannelProcessor(new ChannelProcessor(new NifiChannelSelector(channel)));
+                runner.setSource(source);
+                runner.start();
+            }
+        } catch (Throwable th) {
+            getLogger().error("Error creating source", th);
+            throw Throwables.propagate(th);
+        }
+    }
+
+    @OnUnscheduled
+    public void unScheduled() {
+        if (runner != null) {
+            runner.stop();
+        }
+        if (channel != null) {
+            channel.stop();
+        }
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context,
+            final ProcessSession session) throws ProcessException {
+        if (source instanceof EventDrivenSource) {
+            onEventDrivenTrigger(context, session);
+        } else if (source instanceof PollableSource) {
+            onPollableTrigger((PollableSource) source, context, session);
+        }
+    }
+
+    public void onPollableTrigger(final PollableSource pollableSource,
+            final ProcessContext context, final ProcessSession session)
+            throws ProcessException {
+        try {
+            pollableSource.setChannelProcessor(new ChannelProcessor(
+                    new NifiChannelSelector(new NifiChannel(session, SUCCESS))));
+            pollableSource.start();
+            pollableSource.process();
+            pollableSource.stop();
+        } catch (EventDeliveryException ex) {
+            throw new ProcessException("Error processing pollable source", ex);
+        }
+    }
+
+    public void onEventDrivenTrigger(final ProcessContext context, final ProcessSession session) {
+        Transaction transaction = channel.getTransaction();
+        transaction.begin();
+
+        try {
+            Event event = channel.take();
```

Comment:
Should this get all of the available events from the channel or at least more than one? The event driven source "provides its own event-driven mechanism to invoke event processing". That sounds like it will produce events on this channel without outside intervention and may have produced many between the time that NiFi triggers the processor. If that's the case, then I think this needs to consume as many as possible.